File: day2/sorted_squared_array.py
# Time Complexity: O(n)
# Space Complexity: O(n)
def sortedSquaredArray(array):
    resultList = [0] * len(array)
    leftPointer = 0
    rightPointer = len(array) - 1
    for i in reversed(range(len(array))):
        if abs(array[leftPointer]) < abs(array[rightPointer]):
            resultList[i] = array[rightPointer]**2
            rightPointer -= 1
        else:
            resultList[i] = array[leftPointer]**2
            leftPointer += 1
    return resultList

File: day2/sorted_squared_array.java
/* Time Complexity: O(n)
Space Complexity: O(n) */ 
import java.util.*;

class Program {
  public int[] sortedSquaredArray(int[] array) {
    int[] resultArray = new int[array.length];
    int leftPointer = 0, rightPointer = array.length - 1;
    int indexPointer = array.length - 1;
    for(int i=0;i<array.length;i++) {
      if(Math.abs(array[leftPointer]) < Math.abs(array[rightPointer])) {
        resultArray[indexPointer - i] = array[rightPointer] * array[rightPointer];
        rightPointer--;
      }
      else{
        resultArray[indexPointer - i] = array[leftPointer] * array[leftPointer];
        leftPointer++;
      }
    }
    return resultArray;
  }
}
